### PR TITLE
Assert at least 4 cores at CI environment

### PR DESCRIPTION
**Pull request by blindpirate**
See https://github.com/gradle/gradle-private/issues/3006

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -145,6 +145,8 @@ if (isSkipBuildSrcVerification) {
 }
 
 if (isCiServer) {
+    println("Current machine has ${Runtime.getRuntime().availableProcessors()} CPU cores")
+    require(Runtime.getRuntime().availableProcessors() >= 4) { "CI machines must have at least 4 CPU cores!" }
```
Comment (blindpirate) [inline buildSrc/build.gradle.kts]: Seems like we need to assert more CPU cores on Heztner boxes, any suggestions?

Comment (wolfs) [reply buildSrc/build.gradle.kts]: Do we only have 4 cores on EC2?

How about we assert the 6 cores on Windows only?

Comment (wolfs) [reply buildSrc/build.gradle.kts]: Do we get a build scan when we fail here?

Comment (blindpirate) [reply buildSrc/build.gradle.kts]: That bothers me a lot: we have the following archetectures:

- 8 cores on Hetzner Linux
- 8 cores on EC2 Linux
- 6 cores on Windows
- 4 cores on MacOS

> How about we assert the 6 cores on Windows only?

I'm not sure if that would be problem in the future on other platforms... ok, let's not do premature optimization.

**Comment by blindpirate** [comment]
Good point, I'll make sure we can see such failures in build scan.

**Comment by blindpirate** [comment]
The failure scan: https://e.grdev.net/s/cikkpcrc5ofgu/failure?openFailures=WzBd#top=0

**Pull request merged**